### ISSUE TITLE
How do I set perturbations for each state vector element in HEMCO_Config.rc for global methane inversion in GEOS-Chem 14.3.1?

### ISSUE THREAD

**Issue by XunZhangNJU** [geoschem/geos-chem#2290]
### Name and Institution (Required)

Name: XunZhang
Institution: Nanjing University

### Confirm you have reviewed the following documentation

- [ ] [Support guidelines](https://geos-chem.readthedocs.io/en/stable/help-and-reference/SUPPORT.html)
- [ ] [User manuals](https://geos-chem.readthedocs.io/en/stable/geos-chem-shared-docs/supplemental-guides/related-docs.html)
- [ ] [Debugging GEOS-Chem and HEMCO errors](https://geos-chem.readthedocs.io/en/stable/geos-chem-shared-docs/supplemental-guides/debug-guide.html)
- [ ] [Current and past Github issues](https://github.com/geoschem/geos-chem/issues)

### Description of your issue or question
I am currently conducting a global methane inversion simulation using GEOS-Chem version 14.3.1 and have encountered an issue that I hope you can assist me with. In my HEMCO_Config.rc file, I have set the 'AnalyticalInversion' to true as per the documentation guidelines.
![微信图片_20240513123548](https://github.com/geoschem/geos-chem/assets/22512680/9097582d-d8ad-48d2-962a-096977b657bd)

However, I am uncertain about the correct configuration for the scale factors associated with analytical inversions. Upon running the './gcclassic' command as advised, I encountered an error.
![微信图片_20240513123552](https://github.com/geoschem/geos-chem/assets/22512680/8400d335-f155-4308-8eb4-f3f905e284fd)

![微信图片_20240513124231](https://github.com/geoschem/geos-chem/assets/22512680/a1b4aec3-8366-455e-a134-dc9c4dd9a8cc)

Given that I have a state vector with 3328 elements on a global scale, could you provide guidance on setting up the SCALE_ELEM_000N parameters? If possible, a reference Perturbations.txt file or an example of how to configure these scale factors would be immensely helpful.

Here is my HEMCO_Config.rc and cofig.yml.
[HEMCO_Config.txt](https://github.com/geoschem/geos-chem/files/15289550/HEMCO_Config.txt)
[geoschem_config.txt](https://github.com/geoschem/geos-chem/files/15289551/geoschem_config.txt)

**Comment by yantosca** [comment]
Thanks for writing @XunZhangNJU.  Are you using the IMI workflow?  if so, then this issue might be better submitted to https://github.com/geoschem/integrated_methane_inversioln issue tracker.

Tagging @msulprizio

**Comment by msulprizio** [comment]
Hi @XunZhangNJU. Thanks for your interest in this capability. That option is meant to be used within the Integrated Methane Inversion (https://imi.seas.harvard.edu/) framework that Bob mentions above. It will be included with the IMI 2.0 release slated for release later this summer. The updates in [this pull request](https://github.com/geoschem/integrated_methane_inversion/pull/193) enable the IMI to automatically set up GEOS-Chem run directories to properly apply emissions perturbations to each state vector element. That pull request is being merged into the `dev` branch as we speak and will be included in IMI 2.0.

To utilize the `AnalyticalInversion` capability with GEOS-Chem 14.3.1 and later out-of-the-box (i.e. not within the IMI) will require a fair amount of setup on your end. You will need to create a run directory for each state vector element and provide a netCDF file with the state vector definition. Each run directory would also need a `Perturbations.txt` file as you referenced above. The contents of that should be something like:

```
# Input file for applying emissions perturbations to state vector elements via HEMCO

# Mask field name
CH4_STATE_VECTOR

# Name   | ID  | Scale factor
DEFAULT    0     1.0
ELEM_0001  1     1.5
```

and corresponds to the following lines in HEMCO_Config.rc:

```
#==============================================================================
# --- Scale factors for analytical inversions ---
#==============================================================================
(((AnalyticalInversion

# Add perturbations to individual state vector element (N) following this format
# Start scale factor ID at 2000 to avoid conflicts with other SFs/masks
#200N SCALE_ELEM_000N ./Perturbations.txt - - - xy count 1
2001 SCALE_ELEM_0001 Perturbations_0001.txt - - - xy count 1

)))AnalyticalInversion
```

Again, the IMI will automatically set this up for you based on a provided state vector file (or it can create one for you). I'd recommend checking that out and submitting issues there if you have further questions.

**Comment by XunZhangNJU** [comment]
That’s clear, thanks. @yantosca, @msulprizio.
